### DISCUSSION TITLE
How to access the container's name inside the container?

### DISCUSSION THREAD

**Discussion by Zhenshan-Jin** [opencontainers/runc#3517] [Q&A] [answered]
I am wondering if it is possible to access the container's name inside the container?

For example, I create a container with
```
sudo runc run MY_CONTAINER
```
is it possible to access MY_CONTAINER inside the container?

The use case is that in each of the created container, I will create a unix socket so that the server in the container coud be called by the outside.

Any help is appreciated : )

**Comment by cyphar** [comment, marked as answer]
There is no guaranteed way of getting that information, since the container name is just used by runc for managing its containers.

However (unless the container is configured with custom cgroup configuration) you can find the name of the container as the final component of the cgroup pathnames in `/proc/self/cgroup` inside the container:

```
% runc run -b bundle ctr
# cat /proc/self/cgroup
0::/user.slice/user-1000.slice/ctr
```

(On cgroupv1 systems there will be more cgroup entries.) Please note this is not guaranteed -- it is possible to specify a custom cgroup path which will hide this information.

**Reply by Zhenshan-Jin** [reply]
Thank you @cyphar. I tried the steps but when I do `cat /proc/self/cgroup`, I only see `0::` without other information. Maybe I missed some steps ?

**Reply by cyphar** [reply]
If you're using cgroup namespaces then you won't be able to see anything.

**Reply by Zhenshan-Jin** [reply]
Thank you! @cyphar

**Comment by lifubang** [comment]
How about passing it through an env variable in “config.json”?

**Reply by cyphar** [reply]
`config.json` doesn't have any variable substitution or any other similar feature you'd need to do that. runc is intended to be used as part of a container manager -- if you need this feature, the simplest solution would be to write a script to be written which fills in the relevant environment variable using `jq` before spawning the container.
